src/plonk/prover.rs

```diff
@@ -529,50 +529,70 @@ impl<C: CurveAffine> Proof<C> {
                 })
                 .or_else(|| Some(poly));
         }
-        let mut f_poly = f_poly.unwrap();
-        let mut f_blind = Blind(C::Scalar::random());
 
-        let f_commitment = params.commit(&f_poly, f_blind).to_affine();
+        let final_opening;
+        let final_f_commitment;
+        let final_q_evals;
 
-        hash_point(&mut transcript, &f_commitment)?;
+        loop {
+            let mut transcript_dup = transcript.clone();
```

Comment:
It would be better to do

```rust
let mut transcript = transcript.clone();
let mut transcript_scalar = transcript_scalar.clone();
```

effectively shadowing `transcript`/`transcript_scalar` so that we don't accidentally interact with them inside the loop.